specification/communication/data-plane/Email/preview/2023-15-01-preview/CommunicationServicesEmail.json

```diff
@@ -0,0 +1,413 @@
+                "type": "string"
+              }
+            },
+            "schema": {
+              "$ref": "../../../Common/stable/2021-03-07/common.json#/definitions/CommunicationErrorResponse"
+            }
+          },
+          "202": {
+            "description": "Message accepted, the `Operation-Location` header contains the URL to poll for the message status.",
+            "headers": {
+              "Repeatability-Result": {
+                "description": "Status of a repeatable request.",
+                "type": "string"
+              },
+              "Operation-Location": {
+                "description": "Location url of where to poll the status of this message from.",
+                "type": "string"
+              },
+              "Retry-After": {
+                "description": "Amount of time client should wait before retrying the request, specified in seconds ",
+                "type": "integer"
+              },
+              "x-ms-request-id": {
+                "description": "System generated GUID indicating the id of this specific message, to be used for tracking purposes.",
+                "type": "string"
+              }
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Send Email": {
+            "$ref": "./examples/SendEmail.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "Message": {
+      "description": "Message payload for sending an email",
+      "required": [
+        "senderEmail",
+        "content",
+        "recipients"
+      ],
+      "type": "object",
+      "properties": {
+        "headers": {
+          "description": "Custom email headers to be passed.",
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          }
+        },
+        "senderEmail": {
+          "description": "Sender email address from a verified domain.",
+          "type": "string",
+          "example": "info@contoso.com"
+        },
+        "content": {
+          "description": "Email content to be sent.",
+          "$ref": "#/definitions/EmailContent"
+        },
+        "importance": {
+          "description": "The importance type for the email.",
+          "$ref": "#/definitions/EmailImportance"
+        },
+        "recipients": {
+          "description": "Recipients for the email.",
+          "$ref": "#/definitions/EmailRecipients"
+        },
+        "attachments": {
+          "description": "list of attachments",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/EmailAttachment"
+          }
+        },
+        "replyTo": {
+          "description": "Email addresses where recipients' replies will be sent to.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/EmailAddress"
+          }
+        },
+        "disableUserEngagementTracking": {
```

Comment:
1) Should this rather be enable and off by default?
2) Where do customers get engagement analytics, are there any other REST APIs to query them?

Reply:
UserEngagementTracking is a flag in the portal/control plane at the level of the Email Domains resource. It's off by default there. It can be turned on by the resource owner. This setting is specifically to override the control-plane flag (if that was turned on) on a per-request basis. This flag is a no-op if the control-plane flag is off.  
The idea is that an admin/resource owner sets the user engagement tracking flag at the resource level. Then anyone who wants to override it at the per-request level, can use this flag.

There are currently no REST API to query the engagement analytics. It's published to our customers either through shoebox/Azure Monitor or through EventGrid events.

Reply:
https://learn.microsoft.com/en-us/azure/event-grid/event-schema-communication-services?toc=https%3A%2F%2Flearn.microsoft.com%2Fen-us%2Fazure%2Fcommunication-services%2Ftoc.json&bc=https%3A%2F%2Flearn.microsoft.com%2Fen-us%2Fazure%2Fbread%2Ftoc.json - maybe add a doc here for email events.

Reply:
Control plane - says user interaction tracking, REST API says disableUserEnagagementTracking -> fix this.

Reply:
Filed a task to fix the name mismatch for user interaction tracking. Only the Portal UI specifies this as 'user interaction tracking' whereas the rest of our API/SDK/events use UserEngagementTracking. 
Filed another task to improve documentation for our email events as discussed above.